interface/java_binding/src/test/java/org/khronos/ktx/test/KtxTexture2Test.java

```diff
@@ -227,4 +236,127 @@ public void testCreate() {
 
         texture.destroy();
     }
+
+	@Test
+	public void testInputSwizzleBasisEx() throws IOException {
+
+		// Create RGBA pixels for an image with 32x32 pixels,
+		// filled with
+		// 8 rows of red pixels
+		// 8 rows of green pixels
+		// 8 rows of blue pixels
+		// 8 rows of white pixels
+		int sizeX = 32;
+		int sizeY = 32;
+		byte[] rgba = new byte[sizeX * sizeY * 4];
+		TestUtils.fillRows(rgba, sizeX, sizeY, 0, 8, 255, 0, 0, 255); // Red
+		TestUtils.fillRows(rgba, sizeX, sizeY, 8, 16, 0, 255, 0, 255); // Green
+		TestUtils.fillRows(rgba, sizeX, sizeY, 16, 24, 0, 0, 255, 255); // Blue
+		TestUtils.fillRows(rgba, sizeX, sizeY, 24, 32, 255, 255, 255, 255); // White
+
+		// Create a texture and fill it with the RGBA pixel data
+		KtxTextureCreateInfo info = new KtxTextureCreateInfo();
+		info.setBaseWidth(sizeX);
+		info.setBaseHeight(sizeY);
+		info.setVkFormat(VkFormat.VK_FORMAT_R8G8B8A8_SRGB);
+		KtxTexture2 t = KtxTexture2.create(info, KtxCreateStorage.ALLOC);
+		t.setImageFromMemory(0, 0, 0, rgba);
+
+		// Apply basis compression with an input swizzle, BRGA, so that
+		// the former B channel becomes the R channel
+		// the former R channel becomes the G channel
+		// the former G channel becomes the B channel
+		// the former A channel remains the A channel
+		KtxBasisParams p = new KtxBasisParams();
+		p.setUastc(false);
+		p.setInputSwizzle(new char[] { 'b', 'r', 'g', 'a' });
+		t.compressBasisEx(p);
+
+		// Transcode the resulting texture to RGBA32
+		int outputFormat = KtxTranscodeFormat.RGBA32;
+		int transcodeFlags = 0;
+		t.transcodeBasis(outputFormat, transcodeFlags);
+		byte[] actualRgba = t.getData();
+
+		// Define the expected RGBA pixels. These are the swizzled input
+		// pixels, with slight deviations due to compression artifacts
+		byte[] expectedRgba = new byte[sizeX * sizeY * 4];
+		TestUtils.fillRows(expectedRgba, sizeX, sizeY, 0, 8, 2, 255, 2, 255);
```

Comment:
This is very fragile, subject to breakage when the encoder changes. I don't know how it is working across platforms given the issue I pointed you at yesterday.  I thought the "fixed" solution you were proposing was to also encode the source without swizzling and compare the rows of the transcoded swizzled data with what are expected to be the matching rows of the transcoded unswizzled data.

Reply:
This refers to a change that we talked about in the 'swizzle PR'. (Sorry about the confusion here - I'll try to not open parallel/overlapping PRs in the future). 

I summarized my understanding about how the test should be done at https://github.com/KhronosGroup/KTX-Software/pull/876#issuecomment-2022693341 now.

Reply:
Resolving this as it has been fixed in #876.